src/mon/OSDMonitor.cc

```diff
@@ -3073,11 +3074,26 @@ bool OSDMonitor::preprocess_command(MonOpRequestRef op)
     string format;
     cmd_getval(g_ceph_context, cmdmap, "format", format);
     boost::scoped_ptr<Formatter> f(Formatter::create(format, "json-pretty", "json-pretty"));
-    f->open_object_section("osd_metadata");
-    r = dump_osd_metadata(osd, f.get(), &ss);
-    if (r < 0)
-      goto reply;
-    f->close_section();
+    if (osd >= 0) {
+      f->open_object_section("osd_metadata");
+      r = dump_osd_metadata(osd, f.get(), &ss);
```

Comment:
Not super improtant since they asked for it, but we could include the "id" here just for consistency